backend/tests/api/test_main.py

```diff
@@ -39,6 +39,21 @@ def test_food(self):
 
         # TODO: remove entry from db
 
+    def test_remove_food(self):
+        payload = {
+            "name": "ratata",
+            "weight": 123,
+            "date_produced": "2020-05-10",
+            "expiry_date": "2025-05-10",
+            "decription": "victory hw",
+            "category": "Vegetables",
+            "serving_size": "1",
+        }
+        response = self.app.post("/add_food", json=payload)
+
+        response = self.app.delete("/remove_food/1")
```

Comment:
if your db is not empty, your food_id wont be 1
so this test wouldn't work, right?

Reply:
couple of things you could do instead:

- remove last food

- find a way to query food and get id

- this should be done in crud testing, not route testing

Reply:
The reason I did this way is because since the tests are processing async, so I don't know if there is a food in the DB already. So I called add_food to make sure there is at least 1 food in the DB. Then remove_food is called on id=1, and as long as we're able to remove that food (whether it be the newly added or the old one), it should work
* how do we remove the last food if we are unsure if there is one? how do we get the ID?
* how do you suggest we do that? make a new route called get_all_food? then this would no longer be testing the route but testing CRUD as you said, but we still need a way to test the route independently, right?
* should we make a new file for crud testing for all the routes?

Reply:
`how do we remove the last food if we are unsure if there is one? how do we get the ID?`
you can do a query all and check if that is null or not to check if there is something in the DB.
if the db is not null, you can get the ID
https://docs.sqlalchemy.org/en/13/orm/query.html

`how do you suggest we do that? make a new route called get_all_food? then this would no longer be testing the route but testing CRUD as you said, but we still need a way to test the route independently, right?`
you dont need a route get_all_food, you need a crud method that does that. The whole point for testing a route is to test the response.status not the elements.

`should we make a new file for crud testing for all the routes?`
crud testing is for the **methods** not the routes. The routes use the methods to make an action on the db.

Reply:
please check the latest changes